crates/storage-query-datafusion/src/physical_optimizer.rs

```diff
@@ -53,20 +62,34 @@ impl PhysicalOptimizerRule for JoinRewrite {
                 hash_join.null_equals_null(),
                 None,
                 None,
+                // hash_join
+                //     .left()
+                //     .properties()
+                //     .output_ordering()
+                //     .map(|s| s.to_vec()),
+                // hash_join
+                //     .right()
+                //     .properties()
+                //     .output_ordering()
+                //     .map(|s| s.to_vec()),
                 StreamJoinPartitionMode::Partitioned,
             ) else {
-                return Ok(Transformed::No(plan));
+                return Ok(Transformed::no(plan));
             };
 
-            Ok(Transformed::Yes(Arc::new(new_plan)))
-        })
+            let new_plan: Arc<dyn ExecutionPlan> = Arc::new(new_plan);
+
+            Ok(Transformed::yes(new_plan))
+        });
+
+        res.map(|t| t.data)
     }
 
     fn name(&self) -> &str {
         "join_rewrite"
     }
 
     fn schema_check(&self) -> bool {
-        true
+        false
```

Comment:
Out of curiosity: What is changing the schema so that we need to disable the schema check?